### PR TITLE
Fix/cplp 1187/submit btn issue

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**

### CODE REVIEW

File: src/components/dragdrop.tsx [hunk 1]
```diff
@@ -47,24 +43,28 @@ export const DragDrop = ({
   // Return the current status of files being uploaded
   const handleChangeStatus = ({ meta, file }, stats) => {
     console.log(stats, meta, file)
+    if (stats === 'done' && file.type !== 'application/pdf')
+      toast.error('...')
```
Comment (oyo) [inline src/components/dragdrop.tsx]: please add the according error message here - also the i18n is missing

File: src/components/dragdrop.tsx [hunk 2]
```diff
@@ -47,24 +43,28 @@ export const DragDrop = ({
...
   }
 
   // Return array of uploaded files after submit button is clicked
   const handleSubmit = async (files: IFileWithMeta[], allFiles) => {
+    console.log('allFiles', allFiles)
```
Comment (oyo) [inline src/components/dragdrop.tsx:52]: please remove console logs before PR

File: src/components/dragdrop.tsx [hunk 3]
```diff
@@ -47,24 +43,28 @@ export const DragDrop = ({
   // Return the current status of files being uploaded
   const handleChangeStatus = ({ meta, file }, stats) => {
     console.log(stats, meta, file)
+    if (stats === 'done' && file.type !== 'application/pdf')
+      toast.error('Only .pdf files are allowed')
```
Comment (oyo) [inline src/components/dragdrop.tsx:47]: please use translations here

File: src/components/dragdrop.tsx [hunk 4]
```diff
@@ -47,24 +43,28 @@ export const DragDrop = ({
...
   const handleSubmit = async (files: IFileWithMeta[], allFiles) => {
+    console.log('allFiles', allFiles)
     if (files.length > 2) {
       toast.error('Cannot upload more than two files')
       return
     }
-    files.forEach((document) => dispatch(saveDocument({applicationId, document})))
+    files.forEach((document) => dispatch(saveDocument({ applicationId, document })))
+    dispatch(addFileNames(files.map((file) => file.file.name)))
     toast.success('All files uploaded')
```
Comment (oyo) [inline src/components/dragdrop.tsx:59]: this is not a good error handling - with `dispatch` we only trigger the action. But we don't know if the `saveDocument` call was successful or not. So instead listen to the result of the upload using `useSelector` - then show a message if the upload was successful or not

**Review by oyo — CHANGES_REQUESTED** [review]
some changes required - especially there is no error handling

**Comment by oyo** [comment]
more details about the error handling:
- get the result of the upload action with `useSelector(stateSelector)` from the applicationDocuments slice
- show a success toast when `.uploadRequest` is `'true'` otherwise show an error toast

**Comment by jjeroch** [comment]
> more details about the error handling:
> 
> * get the result of the upload action with `useSelector(stateSelector)` from the applicationDocuments slice
> * show a success toast when `.uploadRequest` is `'true'` otherwise show an error toast

Quick point on this. I dont want a success toast please. We push the user to the next window, so a success message is not needed on top. That would be a double message. Regarding the error, I fully agree

**Comment by nidhigarg-bmw** [comment]
> > more details about the error handling:
> > 
> > * get the result of the upload action with `useSelector(stateSelector)` from the applicationDocuments slice
> > * show a success toast when `.uploadRequest` is `'true'` otherwise show an error toast
> 
> Quick point on this. I dont want a success toast please. We push the user to the next window, so a success message is not needed on top. That would be a double message. Regarding the error, I fully agree
@jjeroch Okay, I have removed the success message and showing only error message if any.

**Review by oyo — APPROVED** [review]
looks good to me now

**Pull request merged**